core/src/main/java/greencity/controller/EcoNewsController.java

```diff
@@ -165,4 +165,30 @@ public EcoNewsController(EcoNewsService ecoNewsService) {
         }
         return ResponseEntity.status(HttpStatus.OK).body(ecoNewsService.find(page, tags));
     }
+
+    /**
+     * Method for getting three eco news for recommendations widget.
+     *
+     * @return list of three recommended {@link EcoNewsDto} instances.
+     * @author Yurii Zhurakovskyi.
+     */
+    @ApiOperation(value = "Get three recommended eco news.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = HttpStatuses.OK),
+            @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+            @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping("/recommended")
+    public ResponseEntity<List<EcoNewsDto>> getThreeRecommendedEcoNews(
+            @ApiParam(value = "Tags for priorities(if do not input tags get without priorities)")
+            @RequestParam(required = false) List<String> tags,
+            @RequestParam(required = true) Long openedEcoNewsId
+    ) {
+        if (tags == null || tags.isEmpty()) {
```

Comment:
- I would move this check to service level
- I would add 'else' to explicitly indicate that those are two exclusive branches of method execution

Reply:
I combined getThreeRecommendedEcoNews, getThreeRecommendedEcoNewsByTags service methods into a single getThreeRecommendedEcoNews service method, moved tag count check to service level